### DISCUSSION TITLE
How to Update CloudFlare API Token for SSL

### DISCUSSION THREAD

**Discussion by exliko** [NginxProxyManager/nginx-proxy-manager#4190] [Q&A] [answered]
I need to update my CloudFlare API Token for the SSL 
But I can't find where is it? I go to SSL Certificates Tab and click 3 dots on a domain, there are only Renew Now, Download and Delete.
How to edit it?

**Comment by lastsamurai26** [comment, marked as answer]
you have 2 ways. 
1. delete and recreate and save all sites again 
2. via ssh -> you need to find out your ssl ID  -> findable at the three dots. 
then go to the folder where letsencrypt is saving the cert 
Like  /opt/containers/nginxproxy/letsencrypt
find your id in the credentials/ folder and change the Token :-) 
restart the npm and try

**Reply by exliko** [reply]
Thx

I can't do the no 1, if I delete it and then recreate it, it always failed, all i have to do is delete first all the hosts that use that SSL, recreate the SSL and recreate the host. Which is stressful, I got 20ish hosts.

But your solution no. 2, works like a charm, it can be done under a minute.

Why NPM doesn't make GUI to update the API Token......... ><
